x-pack/plugins/apm/public/hooks/useEnvironments.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { useFetcher } from './useFetcher';
+import {
+  ENVIRONMENT_NOT_DEFINED,
+  ENVIRONMENT_ALL
+} from '../../common/environment_filter_values';
+import { useUrlParams } from './useUrlParams';
+
+export const ALL_OPTION = {
+  value: ENVIRONMENT_ALL,
+  text: i18n.translate('xpack.apm.environment.allLabel', {
+    defaultMessage: 'All'
+  })
+};
+
+function getEnvironmentOptions(environments: string[]) {
+  const environmentOptions = environments
+    .filter(env => env !== ENVIRONMENT_NOT_DEFINED)
+    .map(environment => ({
+      value: environment,
+      text: environment
+    }));
+
+  return [ALL_OPTION, ...environmentOptions];
+}
+
+export function useEnvironments() {
+  const { urlParams } = useUrlParams();
```

Comment:
at the very least I would like this hook not to have a dependency on `useUrlParams`, but pass in the urlParams. It's unlikely that we can use the `useUrlParams` hook outside of the APM app (or in editing mode).